src/Upgrade/Infrastructure/Processor/Strategy/Composer/Steps/CheckCredentialsStep.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace Upgrade\Infrastructure\Processor\Strategy\Composer\Steps;
+
+use Upgrade\Infrastructure\Dto\Step\StepsExecutionDto;
+use Upgrade\Infrastructure\Processor\Strategy\StepInterface;
+
+class CheckCredentialsStep extends AbstractStep implements StepInterface
+{
+    /**
+     * @param \Upgrade\Infrastructure\Dto\Step\StepsExecutionDto $stepsExecutionDto
+     *
+     * @return \Upgrade\Infrastructure\Dto\Step\StepsExecutionDto
+     */
+    public function run(StepsExecutionDto $stepsExecutionDto): StepsExecutionDto
+    {
+        $this->vsc->validateSourceCodeProviderCredentials();
```

Comment:
Should a method return a `StepsExecutionDto` which holds the validation results?

Reply:
To my opinion, `Validation` method should return void or throw exception. 
Of course, we are able to catch the exception and transform to StepsExecutionDto 

`try {
            $this->vsc->validateSourceCodeProviderCredentials();
        } catch (EnvironmentVariableIsNotDefinedException $exception) {
            $stepsExecutionDto->setIsSuccessful(false);
            $stepsExecutionDto->setOutputMessage($exception->getMessage());
        }`

Reply:
Discussed. Leave it as is

Reply:
Discussed. Decided return stepsExecutionDto instead of exception